test/test_init.py

```diff
@@ -270,3 +333,76 @@ def test_pulse(ctx_factory, dim):
     w = w / np.sqrt(2.0)
     pulse = _make_pulse(amp=amp, r0=r0, w=w, r=nodes)
     assert(discr.norm(pulse - (pulse_check * pulse_check), np.inf) < tol)
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+def test_multilump(ctx_factory, dim):
+    """Test the multi-component lump initializer."""
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PyOpenCLArrayContext(queue)
+
+    nel_1d = 4
+    nspecies = 10
+
+    from meshmode.mesh.generation import generate_regular_rect_mesh
+
+    mesh = generate_regular_rect_mesh(
+        a=(-1.0,) * dim, b=(1.0,) * dim, n=(nel_1d,) * dim
+    )
+
+    order = 3
+    logger.info(f"Number of elements: {mesh.nelements}")
+
+    discr = EagerDGDiscretization(actx, mesh, order=order)
+    nodes = thaw(actx, discr.nodes())
+
+    rho0 = 1.5
+    centers = make_obj_array([np.zeros(shape=(dim,)) for i in range(nspecies)])
+    spec_y0s = np.ones(shape=(nspecies,))
+    spec_amplitudes = np.ones(shape=(nspecies,))
+
+    for i in range(nspecies):
+        centers[i][0] = (.1 * i)
+        spec_y0s[i] += (.1 * i)
+        spec_amplitudes[i] += (.1 * i)
+
+    velocity = np.zeros(shape=(dim,))
+    velocity[0] = 1
+
+    lump = MulticomponentLump(dim=dim, nspecies=nspecies, rho0=rho0,
+                              spec_centers=centers, velocity=velocity,
+                              spec_y0s=spec_y0s, spec_amplitudes=spec_amplitudes)
+
+    lump_soln = lump(t=0, x_vec=nodes)
+    numcvspec = get_num_species(dim, lump_soln)
+    print(f"get_num_species = {numcvspec}")
+
+    assert get_num_species(dim, lump_soln) == nspecies
+
+    cv = split_conserved(dim, lump_soln)
+    assert discr.norm(cv.mass - rho0) == 0.0
+
+    p = 0.4 * (cv.energy - 0.5 * np.dot(cv.momentum, cv.momentum) / cv.mass)
+    exp_p = 1.0
+    errmax = discr.norm(p - exp_p, np.inf)
+    assert len(cv.species_mass) == nspecies
+    species_mass = cv.species_mass
+
+    spec_r = make_obj_array([nodes - centers[i] for i in range(nspecies)])
+    r2 = make_obj_array([np.dot(spec_r[i], spec_r[i]) for i in range(nspecies)])
+    expfactor = make_obj_array([spec_amplitudes[i] * actx.np.exp(- r2[i])
+                                for i in range(nspecies)])
+    exp_mass = make_obj_array([rho0 * (spec_y0s[i] + expfactor[i])
+                               for i in range(nspecies)])
```

Comment:
This bothers me a little (the code seems too similar to the code that it's trying to test). Not exactly sure how to fix it though. Maybe if we end up symbolic-izing the initializer this will be fine.

Reply:
This is an identity test. It is duplicative code, but this bothers me less because the code/test here defines exactly what is expected of the initializer.  If the initializer is broken, then this test will fail.  

If it seems better we *could* create a datafile where the expected state is stored and compare against that - but I don't see how that's effectively different.

Reply:
I'm kind of with @majosm here: I'm not sure I see the value. But I'm also not going to fight it.